adapters/YahooAdapter/YahooAdapter/Native/GADMAdapterYahooNativeAd.m

```diff
@@ -0,0 +1,345 @@
+  // User Settings
+  YASRequestMetadataBuilder *builder = [[YASRequestMetadataBuilder alloc] init];
+
+  // Mediator
+  builder.mediator = [NSString stringWithFormat:@"AdMobYAS-%@", GADMAdapterYahooVersion];
+
+  // Keywords
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  if ([strongConnector userKeywords] != nil && [[strongConnector userKeywords] count] > 0) {
+    builder.userKeywords = [strongConnector userKeywords];
+  }
+  YASAds.sharedInstance.requestMetadata = [builder build];
+
+  // Debug Mode
+  if ([strongConnector testMode]) {
+    YASAds.logLevel = YASLogLevelDebug;
+  } else {
+    YASAds.logLevel = YASLogLevelError;
+  }
+
+  // COPPA
+  if ([strongConnector childDirectedTreatment].boolValue) {
+    NSLog(@"[YahooAdapter] Applying COPPA.");
+    [YASAds.sharedInstance applyCoppa];
+  }
+}
+
+- (NSString *)stringForComponent:(NSString *)componentId {
+  id<YASComponent> component = [_nativeAd component:componentId];
+  if ([component conformsToProtocol:@protocol(YASNativeTextComponent)]) {
+    return ((id<YASNativeTextComponent>)component).text;
+  }
+  return nil;
+}
+
+- (GADNativeAdImage *)imageForComponent:(NSString *)componentId {
+  GADNativeAdImage *GADImage;
+
+  id<YASComponent> component = [_nativeAd component:componentId];
+  if ([component conformsToProtocol:@protocol(YASNativeImageComponent)]) {
+    NSURL *url = (NSURL *)((id<YASNativeImageComponent>)component).URL;
+    if (url) {
+      GADImage = [[GADNativeAdImage alloc] initWithURL:url scale:1.0f];
+    }
+  }
+  return GADImage;
+}
+
+- (nullable NSString *)headline {
+  return [self stringForComponent:@"title"];
+}
+
+- (nullable NSString *)body {
+  return [self stringForComponent:@"body"];
+}
+
+- (nullable NSString *)callToAction {
+  return [self stringForComponent:@"callToAction"];
+}
+
+- (nullable NSString *)advertiser {
+  return [self stringForComponent:@"disclaimer"];
+}
+
+- (nullable NSString *)price {
+  return nil;
+}
+
+- (nullable NSString *)store {
+  return nil;
+}
+
+- (nullable NSArray<GADNativeAdImage *> *)images {
+  GADNativeAdImage *mainImage = [self imageForComponent:@"mainImage"];
+  return mainImage ? @[ mainImage ] : nil;
+}
+
+- (GADNativeAdImage *)icon {
+  return [self imageForComponent:@"iconImage"];
+}
+
+- (nullable UIView *)mediaView {
+  id<YASComponent> component = [_nativeAd component:@"video"];
+  if (component) {
+    id<YASNativeViewComponent> videoComponent = (id<YASNativeViewComponent>)component;
+    if ([videoComponent conformsToProtocol:@protocol(YASNativeVideoComponent)]) {
+      _yahooVideoPlayerView = [[YASVideoPlayerView alloc] init];
+      YASErrorInfo *error =
+          [(id<YASNativeVideoComponent>)videoComponent prepareView:_yahooVideoPlayerView];
+      if (error) {
+        NSLog(@"[YahooAdapter]: Failed to render mediaView video: %@", error);
+      }
+      return _yahooVideoPlayerView;
+    }
+  }
+
+  component = [_nativeAd component:@"mainImage"];
```

Comment:
We understood mediaView is for a video view only - please see the Verizon adapter, and, the 'hasVideoContent' method below.   Based on this, lines 252-265 should be deleted.
The mainImage component is handled in the 'images' method above, and it is loading and displaying.

Reply:
My understanding is that without a call to `prepareView`, the `mainImage` asset will not be rendered, which is reflected on my testing.
If I remove lines `252`-`265`, then the image is not rendered.

For reference, I used the example app in https://github.com/googleads/googleads-mobile-ios-examples to verify on my end.